src/main/scala/edu/ie3/simona/model/em/EmAggregatePowerOpt.scala

```diff
@@ -12,29 +12,38 @@ import edu.ie3.simona.ontology.messages.flex.MinMaxFlexibilityMessage.ProvideMin
 import edu.ie3.util.scala.quantities.DefaultQuantities._
 import squants.Power
 
-/** Aggregates flex reference power with the target of reaching 0kW, while
-  * optionally excluding positive flex potential of PV/WEC from the calculation
+import java.lang.Math.signum
+
+/** Aggregates flex reference power with the goal of not exceed a defined target
+  * limit, while optionally excluding positive flex potential of PV/WEC from the
+  * calculation. If the target limit can't be met, the closes possible operation
+  * point should be chosen.
   *
+  * @param targetPower
+  *   power limit that should be not be exceeded
   * @param curtailRegenerative
   *   Whether to include positive flexibility of PV/WEC in reference sum
   *   calculation
   */
-final case class EmAggregateSelfOpt(curtailRegenerative: Boolean)
-    extends EmAggregateFlex {
+final case class EmAggregatePowerOpt(
+    targetPower: Power = zeroKW,
```

Comment:
Maybe we can find more suitable name for `targetPower`, because for values, except the default value, it behaves more like an interval. Also, in the tests it is sometimes called a `limit` and in the `CHANGELOG` it is called `targetLimit`.

Reply:
I see your point. I startet with limit but it isn't really a limit since we allow to exceed if it's infeaseable. So target was and is still my prefered choice. But I would be happy, if you have a good suggestion. Nevertheless I will harmonise the other limits that might be in there...

Reply:
We could name it something like `targetRange` or `targetPowerRange`, to express that we trying to find a value, that is inside of a set range.

Reply:
Yes, but the value itself is one target value, not a range. Therefor I'll opt for targetPower.

Reply:
But isn't it `±targetPower`?

Reply:
yes, but from users perspective range could be interpreted to give two values, which is not supported.

Reply:
Should we name it `targetPowerAbs` or at least document, that this value is seen as an absolute value?

Reply:
Good idea!